src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -82,6 +84,23 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             if (_portOptions.ConnectionMode == DiagnosticPortConnectionMode.Listen)
             {
+                if (_portOptions.GetClearEndpointOnStartup() &&
+                   !RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
+                    File.Exists(_portOptions.EndpointName))
+                {
+                    // In some circumstances stale files from previous instances of dotnet-monitor cause
+                    // the new instance to fail binding. We need to delete the file in this situation.
+                    try
+                    {
+                        _logger.EndpointDeleteAttempt(_portOptions.EndpointName);
+                        File.Delete(_portOptions.EndpointName);
+                    }
+                    catch (Exception ex)
```

Comment:
Is this a situation where we want to swallow all exceptions? I know we went through this exercise as part of the security review, so I just want to make sure we're still following that pattern (where applicable).

Reply:
Yes, failing to delete the endpoint is not a failure mode of the tool, so all exceptions should be caught and logged. And the logging infra will log all of the exception information.